File: node/src/balance.ts
import type { Transport } from './http.js';
import { throwIfError } from './errorMapping.js';
import { Balance, balanceFromJson } from './models.js';

export class BalanceResource {
  constructor(private readonly transport: Transport) {}

  async get(): Promise<Balance> {
    const { status, body } = await this.transport.request('GET', '/api/v2/clients/balance-details');
    throwIfError(status, body);
    return balanceFromJson(JSON.parse(body).data);
  }
}

File: python/src/kalourmade_arkesel/balance.py
import json

from .error_mapping import raise_for_status
from .http import Transport
from .models import Balance


class BalanceResource:
    def __init__(self, transport: Transport):
        self._transport = transport

    def get(self) -> Balance:
        status, raw = self._transport.request("GET", "/api/v2/clients/balance-details")
        raise_for_status(status, raw)
        return Balance.from_dict(json.loads(raw)["data"])
